### PR TITLE
 refactor(cookie): Warning message when boot with default Erlang cookie

### PR DESCRIPTION
**Pull request by lafirest**
## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] ~~Added tests for the changes~~
- [x] ~~Changed lines covered in coverage report~~
- [x] Change log has been added to `changes/` dir
- [x] ~~For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)~~
- [x] For internal contributor: there is a jira ticket to track this change
       EMQX-8043
- [x] ~~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~~
- [x] ~~In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section~~

### CODE REVIEW

File: rebar.config.erl [hunk 1]
```diff
@@ -203,18 +203,21 @@ overlay_vars(RelType, PkgType, false) ->
...
 %% vars per release type, cloud or edge
-overlay_vars_rel(RelType) ->
-    VmArgs = case RelType of
-                 cloud -> "vm.args";
-                 edge -> "vm.args.edge"
-             end,
+overlay_vars_rel(cloud) ->
+    [ {emqx_default_erlang_cookie, "emqxsecretcookie"}
+    , {vm_args_file, "vm.args"}
+    ] ++ overlay_vars_rel_common(cloud);
```
Comment (zmstone) [inline rebar.config.erl:209]:
```suggestion
    | overlay_vars_rel_common(cloud)];
```

File: rebar.config.erl [hunk 2]
```diff
@@ -203,18 +203,21 @@ overlay_vars(RelType, PkgType, false) ->
...
+overlay_vars_rel(edge) ->
+    [ {vm_args_file, "vm.args.edge"}
+    ] ++ overlay_vars_rel_common(edge).
```
Comment (zmstone) [inline rebar.config.erl:212]:
```suggestion
    | overlay_vars_rel_common(edge)].
```

File: rebar.config.erl [hunk 3]
```diff
@@ -203,18 +203,21 @@ overlay_vars(RelType, PkgType, false) ->
...
-overlay_vars_rel(RelType) ->
-    VmArgs = case RelType of
-                 cloud -> "vm.args";
-                 edge -> "vm.args.edge"
-             end,
+overlay_vars_rel(cloud) ->
+    [ {emqx_default_erlang_cookie, "emqxsecretcookie"}
+    , {vm_args_file, "vm.args"}
+    ] ++ overlay_vars_rel_common(cloud);
+overlay_vars_rel(edge) ->
```
Comment (zmstone) [inline rebar.config.erl:210]: don't see why we should make a difference for this.

**Review by id — APPROVED** [review]

File: changes/v4.3.22-en.md [hunk 4]
```diff
@@ -45,6 +45,8 @@
 - For Rule-Engine resource creation failure, delay before the first retry [#9313](https://github.com/emqx/emqx/pull/9313).
   Prior to this change, the retry delay was added *after* the retry failure.
 
+- Print a warning message when boot with the default (insecure) Erlang cookie [#9340](https://github.com/emqx/emqx/pull/9340).
```
Comment (zmstone) [inline changes/v4.3.22-en.md:48]: create a new version 4.3.23

**Review by savonarola — APPROVED** [review]

**Review by ieQu1 — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**